src/urlencoded.rs

```diff
@@ -87,13 +87,11 @@ fn splits_and_parses_an_url() {
         assert_eq!(store.all("car"), None);
     };
 
-    let raw = "http://www.foo.bar/query/test?foo=bar&message=hello&message=world";
-    t(AbsoluteUri(Url::parse(raw).unwrap()));
+    t(Uri::from_static("http://www.foo.bar/query/test?foo=bar&message=hello&message=world"));
 
-    t(AbsolutePath("/query/test?foo=bar&message=hello&message=world".to_string()));
+    t(Uri::from_static("/query/test?foo=bar&message=hello&message=world"));
 
-    assert_eq!(parse_uri(&Star), Params(HashMap::new()));
-
-    let store = parse_uri(&Authority("host.com".to_string()));
+    let uri = Uri::builder().authority("host.com").build().unwrap();
```

Comment:
Again, do we need to `await` or is this ready immediately?  Or am I way off-base?

Reply:
URIs aren't async, since they don't do any i/o